### DISCUSSION TITLE
how to edit, rerun js or block from every third party, inline or first party on each website

### DISCUSSION THREAD

**Discussion by IntelligenceInside** [uBlockOrigin/uAssets#21605] [Q&A]
I wanted to look, edit and re-run on the fly, or block a specific js on a specific page. The issue is eg. on youtube or anythingelse domain, i want to look at js coming from a particular 3rd party eg gstatic or doubeclick.net and then edit or block or allow a particular script from that particlar 3rd party, i cannot do it. There should be an option to show  js from every first party or inline scr or third party on each website and a facility to edit and re-run or block or re-allow it. Thks.

**Comment by BlazeFTL** [comment]
There Is...Try Using Logger

**Reply by IntelligenceInside** [reply]
Logger. Whats that? i am interested to see this feature i ublock. Perhaps its their hidden !!

**Reply by mapx-** [reply]
read the wiki
https://github.com/gorhill/uBlock/wiki/The-logger

**Reply by IntelligenceInside** [reply]
yes, i have used it. Its not showing up the whole scr, which i want to examine and then allow or  block.

**Reply by BlazeFTL** [reply]
Then You Have To Use Devtools Of Chrome

**Reply by gorhill** [reply]
You can click the link icon when hovering over the url in the logger to load the resource into the code viewer. See <https://github.com/gorhill/uBlock/wiki/The-logger#source-code-viewer>.

**Comment by IntelligenceInside** [comment]
> You can click the link icon when hovering over the url in the logger to load the resource into the code viewer. See https://github.com/gorhill/uBlock/wiki/The-logger#source-code-viewer.

Yes, thats right. I figured out. I will look into capability of code viewer.
